src/nvim/eval.c

```diff
@@ -7102,8 +7102,16 @@ static inline int get_float_arg(typval_T *argvars, float_T *f)
 }
 
 // Apply a floating point C function on a typval with one float_T.
-static inline void float_op_wrapper(typval_T *argvars, typval_T *rettv,
-                                    float_T (*function)(float_T))
+//
+// Some versions of glibc on i386 have an optimization that makes it harder to
+// call math functions indirectly from inside an inlined function, causing
+// compile-time errors with some versions of gcc. We avoid trying to inline
+// float_op_wrapper on that platform.
+#ifndef ARCH_32
+inline
+#endif
+static void float_op_wrapper(typval_T *argvars, typval_T *rettv,
```

Comment:
@justinmk `gendeclarations.lua` does not support function pointer arguments intentionally: I 1) did not want to write a recursive parser pattern and 2) I think that arguments with function pointer type without typedef look ugly and are less usable (e.g. you cannot cast slightly incompatible function to the needed type (usually this is the case when in type one argument is `void *`, while actually in function it is `Type *`)).